### DISCUSSION TITLE
What is the best index type for face recognition tasks? vector length 512. What is the best recall rate/speed compromise

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#34675] [Q&A and General discussion]
I have made a software for face recognition.
currently I use FLAT index. Because it has 100% recall rate
But after reaching 10 millions entries it start working not so fast.

what is the best index for this purpose? 
Like I need close to 100% recall rate especially really close neighbors.
Like I don't really care about vectors that are 100th in the similarity, but I really care about first 10 found entities.

**Comment by martinenkoEduard** [comment]
Maybe someone tried to solve similar problem?

**Reply by xiaofan-luan** [reply]
Usually HNSW is a good choice. You can still get 99% recall or above on top10 retrieval

**Reply by martinenkoEduard** [reply]
> Usually HNSW is a good choice. You can still get 99% recall or above on top10 retrieval

What are good starting parameters of HNSW

**Reply by xiaofan-luan** [reply]
you don't need to setup any parameter for hnsw
